src/promptflow/promptflow/_sdk/_utils.py

```diff
@@ -317,6 +317,28 @@ def _match_env_reference(val: str):
     return name
 
 
+def override_connection_config_with_environment_variable(connections: Dict[str, dict]):
+    """
+    The function will use relevant environment variable to override connection configurations. For instance, if there
+    is a custom connection named 'custom_connection' with a configuration key called 'chat_deployment_name,' the
+    function will attempt to retrieve 'chat_deployment_name' from the environment variable
+    'CUSTOM_CONNECTION_CHAT_DEPLOYMENT_NAME' by default. If the environment variable is not set, it will use the
+    original value as a fallback.
+    """
+
+    for connection_name, connection in connections.items():
+        values = connection.get("value", {})
+        for key, val in values.items():
+            if key in connection.get("secret_keys", []):
```

Comment:
Do not need this condition, as we support override both configs & secrets.

Reply:
updated, I thought connection configs only contain "configs" field.